servers/cromwell/jobs/controllers/jobs_controller.py

```diff
@@ -171,41 +171,30 @@ def query_jobs(body, **kwargs):
     headers = kwargs.get('auth_headers')
     query = QueryJobsRequest.from_dict(body)
     query_page_size = query.page_size or _DEFAULT_PAGE_SIZE
-    # Request more than query.page_size from cromwell since subworkflows will get filtered out
-    page_size = query_page_size * 2
     total_results = get_total_results(query, auth, headers)
-
-    results = []
     offset = page_tokens.decode_offset(query.page_token) or 0
-    page = page_from_offset(offset, page_size)
-    last_page = get_last_page(total_results, page_size)
-
-    while len(results) < query_page_size and page <= last_page:
-        page_from_end = last_page - page + 1
-
-        response = requests.post(
-            _get_base_url() + '/query',
-            json=cromwell_query_params(query, page_from_end, page_size),
-            auth=auth,
-            headers=headers)
-
-        if response.status_code == BadRequest.code:
-            raise BadRequest(response.json().get('message'))
-        elif response.status_code == InternalServerError.code:
-            raise InternalServerError(response.json().get('message'))
-        response.raise_for_status()
-
-        # Only list parent jobs
-        now = datetime.utcnow()
-        jobs_list = [
-            format_job(job, now) for job in response.json()['results']
-            if _is_parent_workflow(job)
-        ]
-        jobs_list.reverse()
-        results.extend(jobs_list)
-        offset = offset + page_size
-        page = page_from_offset(offset, page_size)
+    page = page_from_offset(offset, query_page_size)
+    last_page = get_last_page(total_results, query_page_size)
+    page_from_end = last_page - page + 1
+
+    response = requests.post(
+        _get_base_url() + '/query',
+        json=cromwell_query_params(query, page_from_end, query_page_size),
+        auth=auth,
+        headers=headers)
+
+    if response.status_code == BadRequest.code:
+        raise BadRequest(response.json().get('message'))
+    elif response.status_code == InternalServerError.code:
+        raise InternalServerError(response.json().get('message'))
+    response.raise_for_status()
+
+    now = datetime.utcnow()
+    jobs_list = [format_job(job, now) for job in response.json()['results']]
+    jobs_list.reverse()
 
+    if page == last_page:
+        return QueryJobsResponse(results=jobs_list)
     next_page_token = page_tokens.encode_offset(offset)
```

Comment:
Are there tests for this? I can tell by eye this won't work since offset is never incremented.